### DISCUSSION TITLE
Scaling Django Q2

### DISCUSSION THREAD

**Discussion by mast3rbow** [django-q2/django-q2#113] [Q&A]
Hi team,

I'm keen to get the view on how we should scale Django Q in our applications to meet demands.

For example, I've been running a multi-tenanted application which leverages schedules for each customer to perform a series of tasks like alerts, reminders, checking if items have been completed. 

Keen to understand how we should be calculating minimum performance and I guess creating a bit of a guide to help everyone scale their Django Q to get the best experience of this tooling.

Would keen to hear on people who might have been using the tooling longer than I.

**Comment by GDay** [comment]
It really depends on your setup and your needs. But here are a couple of things to keep in mind:

You can set up multiple clusters (so run multiple instances of `python manage.py qcluster`) which should spread the tasks over multiple containers/servers. Ref: https://django-q2.readthedocs.io/en/master/cluster.html#multiple-clusters

If you want to go more specific, you could set up queues specifically for one or more tasks/schedules. With queues, you can have tasks/schedules run a specific server. Say you have a very resource intensive task, then you can define the cluster on which it should run (a server/container with more resources). Or if you have a high priority task/schedule, then you could also create a cluster for those, so they won't be blocked by lower priority tasks/schedules. Ref: https://django-q2.readthedocs.io/en/master/cluster.html#multiple-queues

Further, if you are having issues with a high cpu/ram usage, then you could try to reduce the polling: https://django-q2.readthedocs.io/en/master/configure.html#poll. But if your are waiting for tasks to be completed, then I wouldn't recommend setting this to a high value. You will likely be waiting longer in those cases. If your tasks are not time sensitive (or run for a longer time), then you are free to change this value to make django-q2 a little bit lighter. Same thing with the `guard_cycle`: https://django-q2.readthedocs.io/en/master/configure.html#guard-cycle
